From 307b56f29d6c9cfca2ce67d5cb09fef81daf8505 Mon Sep 17 00:00:00 2001
From: Caesar Wang <wxt@rock-chips.com>
Date: Tue, 3 Jan 2017 14:33:03 +0800
Subject: [PATCH 03/13] CHROMIUM: arm64: dts: rockchip: add the rev2 support
 for Bob board

As the vmmc for sdmmc is low active on Bob rev0/rev1 board.
The hardware fixes the high active for sdmmc on Bob rev2+ board.

BUG=chrome-os-partner:61496
TEST=boot up Bob

Change-Id: I8b9189f7cd2720759c628ac2952a04d08effff68
Signed-off-by: Caesar Wang <wxt@rock-chips.com>
Reviewed-on: https://chromium-review.googlesource.com/422624
Reviewed-by: Brian Norris <briannorris@chromium.org>
---
 arch/arm64/boot/dts/rockchip/Makefile         |  1 +
 .../boot/dts/rockchip/rk3399-gru-bob-r0.dts   |  7 +--
 .../boot/dts/rockchip/rk3399-gru-bob-r2.dts   | 56 +++++++++++++++++++
 3 files changed, 59 insertions(+), 5 deletions(-)
 create mode 100644 arch/arm64/boot/dts/rockchip/rk3399-gru-bob-r2.dts

diff --git a/arch/arm64/boot/dts/rockchip/Makefile b/arch/arm64/boot/dts/rockchip/Makefile
index 8be0a9271c84..12434b20d68b 100644
--- a/arch/arm64/boot/dts/rockchip/Makefile
+++ b/arch/arm64/boot/dts/rockchip/Makefile
@@ -12,6 +12,7 @@ dtb-$(CONFIG_ARCH_ROCKCHIP) += rk3399-evb.dtb
 dtb-$(CONFIG_ARCH_ROCKCHIP) += rk3399-firefly.dtb
 dtb-$(CONFIG_ARCH_ROCKCHIP) += rk3399-gru-kevin.dtb
 dtb-$(CONFIG_ARCH_ROCKCHIP) += rk3399-gru-bob-r0.dtb
+dtb-$(CONFIG_ARCH_ROCKCHIP) += rk3399-gru-bob-r2.dtb
 dtb-$(CONFIG_ARCH_ROCKCHIP) += rk3399-puma-haikou.dtb
 dtb-$(CONFIG_ARCH_ROCKCHIP) += rk3399-sapphire.dtb
 dtb-$(CONFIG_ARCH_ROCKCHIP) += rk3399-sapphire-excavator.dtb
diff --git a/arch/arm64/boot/dts/rockchip/rk3399-gru-bob-r0.dts b/arch/arm64/boot/dts/rockchip/rk3399-gru-bob-r0.dts
index 4256f6bf6785..04c38cb9affa 100644
--- a/arch/arm64/boot/dts/rockchip/rk3399-gru-bob-r0.dts
+++ b/arch/arm64/boot/dts/rockchip/rk3399-gru-bob-r0.dts
@@ -1,5 +1,5 @@
 /*
- * Google Gru-Bob Rev 0+ board device tree source
+ * Google Gru-Bob Rev 0~1 board device tree source
  *
  * Copyright 2016 Google, Inc
  *
@@ -47,10 +47,7 @@
 
 / {
 	model = "Google Bob";
-	compatible = "google,bob-rev7", "google,bob-rev6",
-		     "google,bob-rev5", "google,bob-rev4",
-		     "google,bob-rev3", "google,bob-rev2",
-		     "google,bob-rev1", "google,bob-rev0",
+	compatible = "google,bob-rev1", "google,bob-rev0",
 		     "google,bob", "google,gru", "rockchip,rk3399";
 };
 
diff --git a/arch/arm64/boot/dts/rockchip/rk3399-gru-bob-r2.dts b/arch/arm64/boot/dts/rockchip/rk3399-gru-bob-r2.dts
new file mode 100644
index 000000000000..9c7b4d8ad961
--- /dev/null
+++ b/arch/arm64/boot/dts/rockchip/rk3399-gru-bob-r2.dts
@@ -0,0 +1,56 @@
+/*
+ * Google Gru-Bob Rev 2+ board device tree source
+ *
+ * Copyright 2016 Google, Inc
+ *
+ * This file is dual-licensed: you can use it either under the terms
+ * of the GPL or the X11 license, at your option. Note that this dual
+ * licensing only applies to this file, and not this project as a
+ * whole.
+ *
+ *  a) This file is free software; you can redistribute it and/or
+ *     modify it under the terms of the GNU General Public License as
+ *     published by the Free Software Foundation; either version 2 of the
+ *     License, or (at your option) any later version.
+ *
+ *     This file is distributed in the hope that it will be useful,
+ *     but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *     MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *     GNU General Public License for more details.
+ *
+ *  Or, alternatively,
+ *
+ *  b) Permission is hereby granted, free of charge, to any person
+ *     obtaining a copy of this software and associated documentation
+ *     files (the "Software"), to deal in the Software without
+ *     restriction, including without limitation the rights to use,
+ *     copy, modify, merge, publish, distribute, sublicense, and/or
+ *     sell copies of the Software, and to permit persons to whom the
+ *     Software is furnished to do so, subject to the following
+ *     conditions:
+ *
+ *     The above copyright notice and this permission notice shall be
+ *     included in all copies or substantial portions of the Software.
+ *
+ *     THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+ *     EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES
+ *     OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+ *     NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
+ *     HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY,
+ *     WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+ *     FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+ *     OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+/dts-v1/;
+#include "rk3399-gru-bob.dtsi"
+
+/ {
+	model = "Google Bob";
+	compatible = "google,bob-rev11", "google,bob-rev10",
+		     "google,bob-rev9", "google,bob-rev8",
+		     "google,bob-rev7", "google,bob-rev6",
+		     "google,bob-rev5", "google,bob-rev4",
+		     "google,bob-rev3", "google,bob-rev2",
+		     "google,bob", "google,gru", "rockchip,rk3399";
+};
-- 
2.17.0

